utils/index.js

```diff
@@ -92,3 +101,26 @@ export const getScope = oauthClientId =>
     getMetaDataByOAuthClientId(oauthClientId).then(metadata =>
         get(metadata, 'scope.scope')
     );
+
+export async function hasScopeOverUser(oauthClientId, userId) {
+    const scope = await getScope(oauthClientId);
+    if (includes(SUPER_SCOPES, scope)) {
+        return true;
+    } else if (scope === SCOPE_TYPE.ADMIN) {
+        const metadata = await getMetaDataByOAuthClientId(oauthClientId);
+        const resources = get(metadata, 'resources', []);
+        return !isEmpty(
```

Comment:
this is duplicated with the excetion of the resourceType that we are validating, 

Let's write a method that accepts the resourceType as a parameter